angular/src/lib/select/select.component.ts

```diff
@@ -489,7 +456,29 @@ export class SelectComponent implements AfterContentChecked, AfterContentInit, C
     }, this.closeDuration);
   }
 
-  selectItem(event, option) {
+  private forceSelectItem(emit: boolean = false) {
```

Comment:
So, this is a cut-and-paste of the code used for handling the enter key (I wanted writeValue to share the same flow).  However, when coming in from writeValue, `emit` will be false so there will be no `onChange` event emitted.